python/cuml/cluster/kmeans.pyx

```diff
@@ -432,19 +432,22 @@ class KMeans(Base):
         Sum of squared distances of samples to their closest cluster center.
         """
 
+        self._check_output_type(X)
+
         cdef uintptr_t input_ptr
         X_m, input_ptr, n_rows, n_cols, dtype = \
             input_to_dev_array(X, order='C', check_dtype=self.dtype,
                                convert_to_dtype=(self.dtype if convert_dtype
                                                  else None),
-                               check_cols=self.n_cols)
+                               check_cols=self.n_cols,
+                               legacy=False)
```

Comment:
For discussion... not sure if it's right... what would you think if instead of passing legacy=False everywhere, we just gave this function a new name like `input_to_cuml_array`? Seems like that would be clear and make it easy to grep for/see legacy calls, but we won't be stuck with legacy=False params cluttering the API forever.